ndcube/ndcollection.py

```diff
@@ -0,0 +1,86 @@
+
+import numpy as np
+
+from ndcube import NDCube
+from ndcube.utils.cube import convert_extra_coords_dict_to_input_format
+
+class NDCollection():
+    def __init__(self, data, labels, wcs):
+        """
+        A class for holding and manipulating a collection of aligned NDCube or NDCubeSequences.
+
+        Data cubes/sequences must be aligned, i.e. have the same WCS and be the same shape.
+
+        Parameters
+        ----------
+        data: sequence of `~ndcube.NDCube` or `~ndcube.NDCubeSequence`
```

Comment:
I get that this makes sense in the future, but it feels misleading in the present. If we are enforcing `data` to be `NDCube` then the `.wcs` makes no sense, we would be better off using the first cube in the lists wcs. If we are requiring `wcs` as an arg, we would be better off requiring arrays here?

Reply:
I'm pretty sure I've removed any dependency on NDCube or NDCubeSequence.  This class currently assumes the WCS's are aligned but doesn't check.  That's something we could add.